### PR TITLE
feat: Add Tee Support

### PR DESCRIPTION
**Pull request by abyss-w**

### CODE REVIEW

File: operations/tee.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+				flag = true
+				return
+			}
+
+			output := string(line)
+			fmt.Print(output)
+
+			inputs = append(inputs, line...)
+		}
+	}()
```
Comment (Xuanwo) [inline operations/tee.go:159]: Why do we need to read as lines?

File: operations/tee.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+		r := bufio.NewReader(os.Stdin)
+		b := make([]byte, partSize)
+
+		for {
+			wg.Add(1)
+
+			taskIndex := index
+
+			n, err := io.ReadFull(r, b)
```
Comment (Xuanwo) [inline operations/tee.go:71]: Do we need to reset `b` here?

Comment (abyss-w) [reply operations/tee.go:71]: There is no need to reset `b`. `b` will be overwritten in the course of use.

File: operations/tee.go (hunk 1 above)
Comment (abyss-w) [reply operations/tee.go:159]: By default, the terminal only sends the input to the application after the user presses enter, so we are actually reading one line at a time.

Comment (Xuanwo) [reply operations/tee.go:159]: Here are some misunderstandings about the shell and stdin process. Read `\n` could **corrupt** user's data which is not allowed. For example, `echo "\nHello\nWorld" | byctl tee example:/xxx`.

We should keep reading stdin without consuming any data until we got an `EOF`. In most cases, `EOF` will be triggered in two ways:

- We are using piping, and the parent process existed, for example: `cat x.md | byctl tee xxx`.
- User can use `Ctrl+D` to send `EOF`

Note the difference between `Ctrl+C` and `Ctrl+D`:

- `Ctrl+C` will trigger an `Interupt`, we need to exit now.
- `Ctrl+D` will trigger an EOF, we need to make sure the data have been uploaded before exiting.

Comment (abyss-w) [reply operations/tee.go:159]: Got it!

File: cmd/byctl/tee.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			return err
+		}
+
+		status, err := os.Stdin.Stat()
+		if err != nil {
+			logger.Error("stat stdin", zap.Error(err))
+			return err
+		}
+
+		if (status.Mode() & os.ModeNamedPipe) != os.ModeNamedPipe {
```
Comment (Xuanwo) [inline cmd/byctl/tee.go:73]: Why do we need to check this?

File: docs/rfcs/82-add-tee-support.md [hunk 4]
```diff
@@ -19,6 +19,13 @@ I propose to add support for `tee` command to `BeyondCTL`. `BeyondCTL` will tee
 byctl tee [command options] [target]
 ```
 
+### Command options
+
+```
+--expect-size  expected size of the input file (default: "128MiB")
```
Comment (Xuanwo) [inline docs/rfcs/82-add-tee-support.md:25]: Will we use `expected-size` or `expect-size`?

File: operations/tee.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+	return errch, nil
+}
+
+func (so *SingleOperator) TeeRun(path string) (err error) {
+	var inputs []byte
+
+	r := bufio.NewScanner(os.Stdin)
+
+	for r.Scan() {
```
Comment (Xuanwo) [inline operations/tee.go:129]: Please read from `os.Stdin` directly without `r.Scan` or similar operations.

File: docs/rfcs/82-add-tee-support.md (hunk 4 above)
Comment (abyss-w) [reply docs/rfcs/82-add-tee-support.md:25]: We will use it for multipart upload, which requires partSize, the default is 128MiB.

**Comment by Xuanwo** [comment]
Mostly LGTM, let's create a test case for it.

File: cmd/byctl/tee_test.go [hunk 6]
```diff
@@ -83,4 +66,48 @@ func TestTee(t *testing.T) {
...
+func TestTeeWithExpectSize(t *testing.T) {
+	if os.Getenv("BEYOND_CTL_INTEGRATION_TEST") != "on" {
+		t.Skipf("BEYOND_CTL_INTEGRATION_TEST is not 'on', skipped")
+	}
+	exec.Command("cat")
+
+	path := setupTee(t)
+	defer tearDownTee(t, path)
+
+	err := app.Run([]string{
```
Comment (Xuanwo) [inline cmd/byctl/tee_test.go:93]: You may need to set `app.Reader`.

Comment (abyss-w) [reply cmd/byctl/tee_test.go:93]: Got it!

File: operations/tee.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+	go func() {
+		// Close partch to inform that all parts have been done.
+		defer close(partch)
+
+		wg := &sync.WaitGroup{}
+		var index int
+
+		if r == nil {
+			r = os.Stdin
```
Comment (Xuanwo) [inline operations/tee.go:53]: We don't need to handle the logic here, it's caller's duty.

**Comment by Xuanwo** [comment]
Others LGTM, great work!

**Comment by abyss-w** [comment]
My local tests are fine, but there is an error here.

**Pull request merged**